src/freenet/client/async/SplitFileFetcherStorage.java

```diff
@@ -177,7 +177,12 @@
     final int cooldownTries;
     /** Cooldown lasts this long for each key. */
     final long cooldownLength;
-    /** Only set if all segments are in cooldown. */
+    /** Separate lock for cooldown operations, which must be serialized. Must be taken *BEFORE*
+     * segment locks. DO NOT TAKE THIS LOCK WHILE HOLDING ANY OTHER LOCK! */
+    private final Object cooldownLock = new Object();
+    /** Only set if all segments are in cooldown. Will be either 0 if not in cooldown, -1 if 
+     * finished, or a time at which the request will become fetchable again. 
+     * LOCKING: Protected by cooldownLock, be careful! */
```

Comment:
How about taking out "be careful!" and maybe even "LOCKING:"? I'd be fine with a blank line between the above and the locking documentation.